### ISSUE TITLE
unexpected exception occurred: System.ArgumentException: An item with the same key has already been added. Key: mscorlib when opening Nuget.Core solution

### ISSUE THREAD

**Issue by abpiskunov** [dotnet/cli#2344]
## Steps to reproduce

1. get repo https://github.com/NuGet/NuGet.Client 
2. Set env variable PROJECTMODELSERVER_TRACE=1
3. remove sdk section from NuGet.Client\src\NuGet.Core\global.json
2. Open Nuget.Core.sln

## Expected  behavior

Restore should start

## Actual behavior

No restore happening, open Nuget.Packaging project and see References node has no framework subfolders.

Reason is exception in ProjectModel.Server (see in the output pane , Project Server Model log ):

info: Microsoft.DotNet.ProjectModel.Server.ProcessingQueue[0]
      OnReceive ((8e242c264fa4410294d4ce0aad7141dd, Initialize, 57) -> {
        "Configuration": "Debug",
        "ProjectFolder": "D:\\git\\ms\\nuget\\NuGet.Client\\src\\NuGet.Core\\NuGet.ProjectManagement",
        "Version": 4
      })
fail: Microsoft.DotNet.ProjectModel.Server.ProjectManager[0]
      A unexpected exception occurred: System.ArgumentException: An item with the same key has already been added. Key: mscorlib
         at System.ThrowHelper.ThrowAddingDuplicateWithKeyArgumentException(Object key)
         at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
         at System.Linq.Enumerable.ToDictionary[TSource,TKey](List`1 source, Func`2 keySelector, IEqualityComparer`1 comparer)
         at System.Linq.Enumerable.ToDictionary[TSource,TKey](IEnumerable`1 source, Func`2 keySelector, IEqualityComparer`1 comparer)
         at Microsoft.DotNet.ProjectModel.Server.Helpers.DependencyTypeChangeFinder.Diagnose(ProjectContext context, IEnumerable`1 currentSearchPaths)
         at Microsoft.DotNet.ProjectModel.Server.ProjectContextSnapshot.Create(ProjectContext context, String configuration, IEnumerable`1 currentSearchPaths)
         at Microsoft.DotNet.ProjectModel.Server.ProjectSnapshot.Create(String projectDirectory, String configuration, WorkspaceContext workspaceContext, IReadOnlyList`1 projectSearchPaths)
         at Microsoft.DotNet.ProjectModel.Server.ProjectManager.UpdateProject()
         at Microsoft.DotNet.ProjectModel.Server.ProjectManager.DoProcessLoop()
         at Microsoft.DotNet.ProjectModel.Server.ProjectManager.ProcessLoop()
info: Microsoft.DotNet.ProjectModel.Server.ProcessingQueue[0]
      OnSend ((8e242c264fa4410294d4ce0aad7141dd, Error, 4) -> {
        "Message": "An item with the same key has already been added. Key: mscorlib",
        "Path": null,
        "Line": 0,
        "Column": 0
      })

**Note: project.json for Nuget.Packaging has mscorlib for net45 target as framework assembly, which seems to be a valid case. Even if it is not valid, PMS never should throw this way, since this prevent whole solution to work properly:  Nuget.Packaging project is not initialized correctly, not added to intellisense and all projects depending on it contain errors; build is blocked  etc **

**Note: this exception is not originated from ProjectMOdel server directly, however PMS provides data tooling depends on and never should throw(!), but rather return some error diagnostics.**

## Environment data
`dotnet --info` output:

D:\git\ms\nuget\NuGet.Client>dotnet --info
.NET Command Line Tools (1.0.0-beta-002174)

Product Information:
 Version:     1.0.0-beta-002174
 Commit Sha:  4c22900497

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.10586
 OS Platform: Windows
 RID:         win10-x64

**Comment by abpiskunov** [comment]
For quick reference project.json and lock file are here: "\\vwdbuild01\Temp\antonpis\repro\Nuget.Packaging - dotnet" . Note: PMS throws even when there no lock.

**Comment by troydai** [comment]
Following project.json is being processed when this issue happened:
```
{
  "version": "3.4.0-*",
  "description": "NuGet's implementation for reading nupkg package and nuspec package specification files.",
  "authors": [
    "NuGet"
  ],
  "copyright": "Copyright .NET Foundation. All rights reserved.",
  "projectUrl": "https://github.com/NuGet/NuGet.Packaging/",
  "licenseUrl": "https://raw.githubusercontent.com/NuGet/NuGet.Packaging/master/LICENSE",
  "tags": [
    "semver",
    "semantic versioning"
  ],
  "compilationOptions": {
    "warningsAsErrors": true
  },
  "dependencies": {
    "NuGet.Common": "3.4.0-*",
    "NuGet.Frameworks": "3.4.0-*",
    "NuGet.Shared": {
      "type": "build",
      "version": "3.4.0-*"
    },
    "NuGet.Packaging.Core": "3.4.0-*",
    "NuGet.Logging": "3.4.0-*",
    "NuGet.Versioning": "3.4.0-*"
  },
  "frameworks": {
    "net45": {
      "frameworkAssemblies": {
        "mscorlib": "",
        "System.Xml": "",
        "System.Xml.Linq": "",
        "System.IO.Compression": ""
      }
    },
    "dnxcore50": {
      "dependencies": {
        "System.ObjectModel": "4.0.10-beta-23109",
        "System.IO.Compression": "4.0.0-beta-23109",
        "System.IO.FileSystem": "4.0.0-beta-23109",
        "System.Xml.XDocument": "4.0.10-beta-23109",
        "System.Runtime.Extensions": "4.0.10-beta-23109",
        "System.Reflection.Extensions": "4.0.0-beta-23109",
        "System.Security.Cryptography.Hashing.Algorithms": "4.0.0-beta-23225"
      }
    }
  }
}
```
Look at the file I notice `System.IO.Compress` presents at two target framework. This likely trigger an exception here: https://github.com/dotnet/cli/blob/rel/1.0.0/src/dotnet/commands/dotnet-projectmodel-server/Helpers/DependencyTypeChangeFinder.cs#L20.

I'm now recreating the scenario.

**Comment by troydai** [comment]
I'm able to repo this issue in automation.

```
fail: Microsoft.DotNet.ProjectModel.Server.ProjectManager[0]
      A unexpected exception occurred: System.ArgumentException: An item with the same key has already been added. Key: mscorlib
         at System.ThrowHelper.ThrowAddingDuplicateWithKeyArgumentException(Object key)
         at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
         at System.Linq.Enumerable.ToDictionary[TSource,TKey](List`1 source, Func`2 keySelector, IEqualityComparer`1 comparer)
         at System.Linq.Enumerable.ToDictionary[TSource,TKey](IEnumerable`1 source, Func`2 keySelector, IEqualityComparer`1 comparer)
         at Microsoft.DotNet.ProjectModel.Server.Helpers.DependencyTypeChangeFinder.Diagnose(ProjectContext context, IEnumerable`1 currentSearchPaths)
         at Microsoft.DotNet.ProjectModel.Server.ProjectContextSnapshot.Create(ProjectContext context, String configuration, IEnumerable`1 currentSearchPaths)
         at Microsoft.DotNet.ProjectModel.Server.ProjectSnapshot.Create(String projectDirectory, String configuration, WorkspaceContext workspaceContext, IReadOnlyList`1 projectSearchPaths)
         at Microsoft.DotNet.ProjectModel.Server.ProjectManager.UpdateProject()
         at Microsoft.DotNet.ProjectModel.Server.ProjectManager.DoProcessLoop()
         at Microsoft.DotNet.ProjectModel.Server.ProjectManager.ProcessLoop()
Error => {
  "Message": "An item with the same key has already been added. Key: mscorlib",
  "Path": null,
  "Line": 0,
  "Column": 0
}
```

**Comment by troydai** [comment]
The duplicate libraries keys are fixed in 3d157d0ec8c4f3b98bb4439a0d87366727742edc.
